keras/utils/data_utils.py

```diff
@@ -561,18 +561,23 @@ def _wait_queue(self):
 
     def _run(self):
         """Submits request to the executor and queue the `Future` objects."""
-        sequence = list(range(len(self.sequence)))
-        self._send_sequence()  # Share the initial sequence
         while True:
+            # get new sequence indexes for each epoch
+            sequence = list(range(len(self.sequence)))
+            # Update pool with new sequence
+            self._send_sequence()
```

Comment:
We need to update sequence and sent to the pool at the beginning of each epoch. This is done to make sure correct indexes are used in case if Sequence changes it's length.

Reply:
Maybe add a comment to the code saying this too?